File: backend/stock-serv/src/main/java/org/willingfish/stock/serv/controller/DealingController.java
package org.willingfish.stock.serv.controller;

import org.springframework.web.bind.annotation.GetMapping;
import org.willingfish.stock.serv.pojo.DealingRecord;

import java.util.List;


public interface DealingController {

    @GetMapping("/dealings")
    List<DealingRecord> get();
}

File: backend/stock-serv/src/main/kotlin/org/willingfish/stock/serv/controller/DealingController.kt
package org.willingfish.stock.serv.controller

import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController
import org.willingfish.stock.serv.pojo.DealingRecord
import reactor.core.publisher.Flux

@RestController
@RequestMapping("/dealing")
interface DealingController {

    @GetMapping("/dealings")
    fun dealings(): Flux<DealingRecord>
}